src/ray/raylet/node_manager.cc

```diff
@@ -1231,21 +1232,10 @@ void NodeManager::TreatTaskAsFailed(const Task &task) {
     // information about the TaskSpecification implementation.
     num_returns -= 1;
   }
+  const std::string meta = std::to_string(static_cast<int>(error_type));
   for (int64_t i = 0; i < num_returns; i++) {
     const ObjectID object_id = spec.ReturnId(i);
-
-    std::shared_ptr<Buffer> data;
-    // TODO(ekl): this writes an invalid arrow object, which is sufficient to
-    // signal that the worker failed, but it would be nice to return more
-    // detailed failure metadata in the future.
-    arrow::Status status =
-        store_client_.Create(object_id.to_plasma_id(), 1, NULL, 0, &data);
-    if (!status.IsPlasmaObjectExists()) {
-      // TODO(rkn): We probably don't want this checks. E.g., if the object
-      // store is full, we don't want to kill the raylet.
-      RAY_ARROW_CHECK_OK(status);
-      RAY_ARROW_CHECK_OK(store_client_.Seal(object_id.to_plasma_id()));
-    }
+    RAY_IGNORE_EXPR(store_client_.CreateAndSeal(object_id.to_plasma_id(), "", meta));
```

Comment:
What will happen in this case if the object does not fit? Presumably the driver application will hang? We should at least log a scary message to stderr, but we should probably push a warning to the driver using `gcs_client_->error_table().PushErrorToDriver`.

Reply:
good point, I'll fix this.